core/templates/pages/topic-editor-page/services/topic-editor-state.service.ts

```diff
@@ -18,541 +18,556 @@
+          error || 'There was an error when loading the topic.');
+        this._topicIsLoading = false;
+      });
+    this.topicRightsBackendApiService.fetchTopicRights(
+      topicId).then((newBackendTopicRightsObject) => {
+      this._updateTopicRights(newBackendTopicRightsObject);
+      this._topicIsLoading = false;
+    }, (error) => {
+      this.alertService.addWarning(
+        error ||
+            'There was an error when loading the topic rights.');
+      this._topicIsLoading = false;
+    });
+  }
 
-      /**
-       * Returns whether the topic URL fragment already exists on the server.
-       */
-      getTopicWithUrlFragmentExists: function() {
-        return _topicWithUrlFragmentExists;
-      },
+  getGroupedSkillSummaries(): unknown {
+    return cloneDeep(this._groupedSkillSummaries);
+  }
 
-      /**
-       * Loads, or reloads, the subtopic page stored by this service given a
-       * specified topic ID and subtopic ID.
-       */
-      loadSubtopicPage: function(topicId, subtopicId) {
-        var subtopicPageId = _getSubtopicPageId(topicId, subtopicId);
-        if (_getSubtopicPageIndex(subtopicPageId) !== null) {
-          _subtopicPage = angular.copy(
-            _cachedSubtopicPages[_getSubtopicPageIndex(subtopicPageId)]);
-          _subtopicPageLoadedEventEmitter.emit();
-          return;
-        }
-        EditableTopicBackendApiService.fetchSubtopicPage(
-          topicId, subtopicId).then(
-          function(newBackendSubtopicPageObject) {
-            _updateSubtopicPage(newBackendSubtopicPageObject);
-            $rootScope.$applyAsync();
-          },
-          function(error) {
-            AlertsService.addWarning(
-              error || 'There was an error when loading the topic.');
-          });
-      },
+  getSkillQuestionCountDict(): { [skillId: string]: number; } | undefined[] {
+    return this._skillQuestionCountDict;
+  }
 
-      /**
-       * Returns whether this service is currently attempting to load the
-       * topic maintained by this service.
-       */
-      isLoadingTopic: function() {
-        return _topicIsLoading;
-      },
+  /**
+   * Returns whether the topic name already exists on the server.
+   */
+  getTopicWithNameExists(): boolean {
+    return this._topicWithNameExists;
+  }
 
-      /**
-       * Returns whether a topic has yet been loaded using either
-       * loadTopic() or setTopic().
-       */
-      hasLoadedTopic: function() {
-        return _topicIsInitialized;
-      },
+  /**
+   * Returns whether the topic URL fragment already exists on the server.
+   */
+  getTopicWithUrlFragmentExists(): boolean {
+    return this._topicWithUrlFragmentExists;
+  }
 
-      getSkillIdToRubricsObject: function() {
-        return _skillIdToRubricsObject;
+  /**
+   * Loads, or reloads, the subtopic page stored by this service given a
+   * specified topic ID and subtopic ID.
+   */
+  loadSubtopicPage(topicId: string, subtopicId: number): void {
```

Comment:
This does an async operation and therefore we had used a `$rootScope.$applyAsync();`. but that has been removed. How are we handling this now?

Reply:
Made the function to return a promise. Every angularJS file where this function was called, I have added an $rootScope.$applyAsync();. This removes the requirement of using $rootScope.$applyAsync(); in the function itself